crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -138,9 +150,224 @@ fn faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() -> anyhow::Res
     Ok(())
 }
 
+// TESTS FOR NEW FAUCET EXECUTION ENVIRONMENT
+// ================================================================================================
+
+#[test]
+fn prove_new_faucet_execution_mint_fungible_asset_succeeds() -> anyhow::Result<()> {
+    // CONSTRUCT AND EXECUTE TX using new faucet that exercises the storage offset validation
+    // --------------------------------------------------------------------------------------------
+    let mut builder = MockChain::builder();
+    let faucet = builder.add_existing_faucet(Auth::BasicAuth, "TST", 200, None)?;
+    let mock_chain = builder.build()?;
```

Comment:
I think I don't see what this test is doing differently than `prove_faucet_contract_mint_fungible_asset_succeeds`.

I think we should use `MockChainBuilder::create_new_faucet` here so that we're creating a new faucet which will run the validations that are mentioned in https://github.com/0xMiden/miden-base/issues/912.
I think the same applies to the other newly added tests.